cirq/contrib/acquaintance/gates.py

```diff
@@ -311,5 +311,11 @@ def permutation(self, qubit_count: int) -> Dict[int, int]:
         return {i: j for i, j in
                 enumerate(reversed(range(sum(self.part_lens))))}
 
+    def __repr__(self):
+        return ('cirq.contrib.acquaintance.SwapNetworkGate('
+                '{!r}, {!r})'.format(self.part_lens, self.acquaintance_size))
 
+    def __eq__(self, other):
```

Comment:
You also need `__ne__` and `__hash__`. And the type check.

Alternatively, you can use the `cirq.value_equality` decorator and implement `_value_equality_values_(self)`.

Reply:
Done.